pkg/resource/parameter_group/hooks.go

```diff
@@ -158,6 +159,141 @@ func (rm *resourceManager) resetAllParameters(
 	return &resource{ko}, nil
 }
 
+// customUpdate overrides sdkUpdate by custom logic
+func (rm *resourceManager) customUpdate(
+	ctx context.Context,
+	desired *resource,
+	latest *resource,
+	delta *ackcompare.Delta,
+) (updated *resource, err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.sdkUpdate")
+	defer func() {
+		exit(err)
+	}()
+	if delta.DifferentAt("Spec.Tags") {
+		err = rm.updateTags(ctx, desired, latest)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if delta.DifferentAt("Spec.ParameterNameValues") {
+		ko, err := rm.resetParameterGroup(ctx, desired, latest)
+
+		if ko != nil || err != nil {
+			return ko, err
+		}
+	}
+
+	if !delta.DifferentExcept("Spec.Tags") {
+		return desired, nil
+	}
+
+	input1, input2 := rm.updateRequestPayload(desired)
```

Comment:
I don't think this will scale. Although there are only 39 parameters for now, there may be more later. Rather than update this code again, I would rather we write it with no limits on the maximum number of updates.

Reply:
Updated to a for loop.